File: python/p2384_largest_palindromic_number.py
from collections import Counter

class Solution:
    def largestPalindromic(self, num: str) -> str:
        digit_count = Counter(num)
        res = ''.join(digit_count[c]//2 * c for c in '9876543210').lstrip('0')
        mid = max(digit_count[c]%2 * c for c in digit_count)
        return (res+mid+res[::-1]) or '0'

File: src/solutions/p2384_largest_palindromic_number.rs
use std::collections::HashMap;

pub struct Solution {}

impl Solution {
    pub fn largest_palindromic(num: String) -> String {
        let letter_counts: HashMap<char, usize> = num
            .chars()
            .fold(HashMap::new(), |mut hash_map, c| {
                  *hash_map.entry(c).or_insert(0) += 1;
                  hash_map
            });
        let res = String::new();
        let mut left_string = String::new();
        let mut right_string = String::new();
        for n in (0..=9).rev() {
            let c = char::from_digit(n, 10).unwrap();
            match letter_counts.get(&c) {
                Some(&v) => {
                    if c == '0' && left_string.is_empty() {
                        continue;
                    }
                    while v > 1 {
                        left_string.push(c);
                        right_string.push(c);
                        v -= 2;
                    }
                },
                None => {},
            };
        }
        for n in (0..=9).rev(){
            break;
        }
        return "0".to_string();
    }
}
